src/general_nn_pde_solve.jl

```diff
@@ -0,0 +1,170 @@
+struct NNGeneralPDE{C,O,K} <: NeuralNetDiffEqAlgorithm
```

Comment:
I think the `General` can be dropped from these to maintain consistency https://github.com/SciML/NeuralNetDiffEq.jl/blob/d6831a0cb5a566b244e4e8de8b378ed7aef68830/src/ode_solve.jl#L1 and it always makes sense to add on terms for additional methods and use the generic version for the default?

Reply:
Same for the problem struct?

Reply:
thanks, yeah, it is better here use the generic version "NNDE{C,O,P,K}" for both solvers.